onnx/defs/tensor/defs.cc

```diff
@@ -1185,4 +1185,122 @@ ONNX_OPERATOR_SET_SCHEMA(
+    dimension will be inserted as the innermost dimension, i.e. axis=-1. The size of the additional 
+    dimension is specified by required scalar input 'depth'. The type of the output tensor is the same 
+    as the type of the 'values' input. Any entries in the 'indices' input tensor with values outside 
+    the range [0, depth) will result in one-hot representation with all 'off_value' values in the 
+    output tensor.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    OneHot,
+    9,
+    OpSchema()
+        .SetDoc(OneHot_ver9_doc)
+        .Attr(
+            "axis",
+            "(Optional) Axis along which one-hot representation in added. Default: axis=-1.",
+            AttributeProto::INT,
+            static_cast<int64_t>(-1))
+        .Input(
+            0,
+            "indices",
+            "Input tensor containing indices. The values must be non-negative integers. "
+            "Any entries in the 'indices' input tensor with values outside the range [0, depth) "
+            "will result in one-hot representation with all 'off_value' values in the output tensor."
+            "In case 'indices' is of non-integer type, the values will be casted to int64 before use.",
+            "T1")
+        .Input(
+            1,
+            "depth",
+            "Scalar specifying the number of classes in one-hot tensor. This is also the size "
+            "of the one-hot dimension (specified by 'axis' attribute) added on in the output "
+            "tensor and the values in the 'indices' input tensor are expected to be "
+            "in the range [0, depth). The"
+            "In case 'depth' is of non-integer type, it will be casted to int64 before use.",
+            "T1")
+        .Input(
+            2,
+            "values",
+            "Rank 1 tensor containing exactly two elements, in the format [off_value, on_value], "
+            "where 'on_value' is the value used for filling locations specified in 'indices' input "
+            "tensor, and 'off_value' is the value used for filling locations other than those specified "
+            "in 'indices' input tensor. ",
+            "T2")
+        .Output(0,
+            "output",
+            "Tensor of rank one greater than input tensor 'indices', i.e. rank(output) = rank(indices) + 1. "
+            "The data type for the elements of the output tensor is the same as the type of input 'values' "
+            "is used.",
+            "T2")
+        .TypeConstraint(
+            "T1",
+            OpSchema::all_numeric_types(),
+            "Constrains input to only numeric types.")
+        .TypeConstraint(
+            "T2",
+            OpSchema::all_tensor_types(),
+            "Constrain to any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // Check that the node has two or three onputs.
+          if (!hasNInputShapes(ctx, 3)) {
+            fail_shape_inference(
```

Comment:
Probably not throwing exception, just return to stop the shape inference.

Reply:
Agreed. I have updated the shape inference at multiple places to return whenever there isn't enough info, instead of throwing.